pkg/kubelet/status/status_manager.go

```diff
@@ -335,6 +354,10 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 		klog.Errorf("Status update on pod %v/%v aborted: %v", pod.Namespace, pod.Name, err)
 		return false
 	}
+	if pod.Spec.RestartPolicy != v1.RestartPolicyAlways && oldStatus.Phase == v1.PodFailed && status.Phase != v1.PodFailed {
```

Comment:
If I'm understanding correctly, this functionality seems fairly different from the other functionality to `Fill ContainerStateTerminated with value from kubecontainer.PodStatus`.

Thoughts on adding it via a separate PR?

Reply:
Additionally, could we please provide a unit test verifying this transition is no longer allowed?

Reply:
It seems #58711 points to openshift e2e test which may fail.

I want to see if that is the only way of reproducing the error.

If the change above doesn't fix the e2e test, I will pull it out of this PR.